src/pip/_internal/operations/prepare.py

```diff
@@ -179,7 +179,10 @@ def unpack_url(
 
 
 def _check_download_dir(
-    link: Link, download_dir: str, hashes: Optional[Hashes]
+    link: Link,
+    download_dir: str,
+    hashes: Optional[Hashes],
+    warn_on_hash_mismatch: bool = True,
```

Comment:
Not some that needs to change, I’m just suddenly curious how peopel choose between positive and negative flags. Personally I’d probably choose to implement this as `suppress_hash_mismatch_warning=False`.

Reply:
In this case I think I wanted to avoid a double negation and save half a brain cycle when reading the only condition using this flag.